great_expectations/rule_based_profiler/parameter_builder/value_set_multi_batch_parameter_builder.py

```diff
@@ -0,0 +1,130 @@
+    this ParameterBuilder so please be aware that testing large columns with
+    high cardinality could require a large amount of memory.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        metric_domain_kwargs: Optional[Union[str, dict]] = None,
+        metric_value_kwargs: Optional[Union[str, dict]] = None,
+        data_context: Optional["DataContext"] = None,  # noqa: F821
+        batch_list: Optional[List[Batch]] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+    ):
+        """
+
+        Args:
+            name: the name of this parameter -- this is user-specified parameter name (from configuration);
+            it is not the fully-qualified parameter name; a fully-qualified parameter name must start with "$parameter."
+            and may contain one or more subsequent parts (e.g., "$parameter.<my_param_from_config>.<metric_name>").
+            metric_domain_kwargs: used in MetricConfiguration
+            metric_value_kwargs: used in MetricConfiguration
+            data_context: DataContext
+            batch_list: explicitly passed Batch objects for parameter computation (take precedence over batch_request).
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+        super().__init__(
+            name=name,
+            metric_name="column.distinct_values",
+            data_context=data_context,
+            batch_list=batch_list,
+            batch_request=batch_request,
+            reduce_scalar_metric=False,
+            enforce_numeric_metric=False,
+        )
+
+        self._metric_domain_kwargs = metric_domain_kwargs
+        self._metric_value_kwargs = metric_value_kwargs
+
+    """
+    Full getter/setter accessors for needed properties are for configuring MetricMultiBatchParameterBuilder dynamically.
+    """
+
+    @property
+    def metric_domain_kwargs(self) -> Optional[Union[str, dict]]:
+        return self._metric_domain_kwargs
+
+    @property
+    def metric_value_kwargs(self) -> Optional[Union[str, dict]]:
+        return self._metric_value_kwargs
+
+    @metric_value_kwargs.setter
+    def metric_value_kwargs(self, value: Optional[Union[str, dict]]) -> None:
+        self._metric_value_kwargs = value
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ) -> ParameterContainer:
+
+        # Build the list of unique values for each batch
+        super()._build_parameters(
+            parameter_container=parameter_container,
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+
+        # Retrieve and replace the list of unique values for each batch with
```

Comment:
@anthonyburdi I am thinking of a small refactor focused on making the code segment below dealing with setting the parameter to be an interface that all `ParameterBuilder` classes implement (after a few current PRs get merged).

Reply:
Agree - that makes a lot of sense to me!

Reply:
@anthonyburdi and @alexsherstinsky discussed and agreed to add this in a separate work item.